enhancements/network/user-defined-network-segmentation.md

```diff
@@ -521,13 +520,25 @@ type UserDefinedNetworkSpec struct {
     // Supported by Topology `Layer2` and `Localnet`.
     // +optional
     IPAMLifecycle NetworkIPAMLifecycle `json:"ipamLifecycle,omitempty"`
+    
+    // Control how much of the IP/MAC-address configuration will be managed by OVN.
+    // `Full` means OVN-Kubernetes will apply IP configuration to the SDN infrastructure and it will also assign IPs from the selected subnet to the
+    // individual pods.
+    // `MAC` means OVN-Kubernetes will only assign MAC addresses and provide layer 2 communication, letting users configure IP addresses for the pods. 
```

Comment:
We know that we will later want configuration, where IPs are assigined to the SDN infrastructure, but not to the individual pods.

Could we limit the AddressManagement attribute to the act of assigning an address to the Pod? We would need a different attribute to assign IP to the SDN infrastructure. Some users may want totally unmanaged netowork, some may want managed IPs, but no egress, others may want unmanaged IPs and egress, ...

Reply:
> We know that we will later want configuration, where IPs are assigined to the SDN infrastructure, but not to the individual pods.

What does this mean ? Is this the scenario we've discussed where the GW router exists ? i.e. GW and management port IPs are defined - and the network must have a defined subnet ? 

> Could we limit the AddressManagement attribute to the act of assigning an address to the Pod? We would need a different attribute to assign IP to the SDN infrastructure. Some users may want totally unmanaged netowork, some may want managed IPs, but no egress, others may want unmanaged IPs and egress

I guess sure, but then, what would the values be ? Hard to see this as something other than a boolean.

Reply:
> What does this mean ? Is this the scenario we've discussed where the GW router exists ? i.e. GW and management port IPs are defined - and the network must have a defined subnet ?

Yes, that

> I guess sure, but then, what would the values be ? Hard to see this as something other than a boolean.

The name is more important than the value IMHO.

I would go for "named" true and false, so we could extend it if needed. e.g. IpAddressAssignment: Disabled|Enabled. So later, if needed, we could deprecated Enabled and replace it with 2 or more of more specific names for two specific strategies?

Reply:
Makes sense to me.